libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/RouteOptionsUpdater.kt

```diff
@@ -42,16 +43,46 @@ class RouteOptionsUpdater {
         routeProgress: RouteProgress?,
         locationMatcherResult: LocationMatcherResult?,
     ): RouteOptionsResult {
+        var routeOptions = routeOptions
         if (routeOptions == null || routeProgress == null || locationMatcherResult == null) {
             val msg = "Cannot combine RouteOptions, invalid inputs. routeOptions, " +
                 "routeProgress and locationMatcherResult cannot be null"
             logE(msg, LOG_CATEGORY)
             return RouteOptionsResult.Error(Throwable(msg))
         }
 
-        val coordinatesList = routeOptions.coordinatesList()
+        if (routeOptions.isEVRoute() && routeOptions.waypointIndices() != null) {
+            if (!allCoordinatesAreWaypoints(routeOptions)) {
+                return RouteOptionsResult.Error(
+                    Throwable("Silent waypoints aren't supported in EV routing")
+                )
+            } else {
+                // getting rid of waypoints indices to simplify the rest of the logic
+                routeOptions = routeOptions.toBuilder().waypointIndicesList(null).build()
+            }
+        }
 
-        val (nextCoordinateIndex, remainingCoordinates) = indexOfNextRequestedCoordinate(
+        val currentNavigationRoute = routeProgress.navigationRoute
+        val serverAddedChargingStationWithIndex =
+            currentNavigationRoute.internalWaypoints().mapIndexedNotNull { index, waypoint ->
+                val wasAlreadyRequestedAsUserProvided = currentNavigationRoute.waypoints
+                    ?.getOrNull(index)
```

Comment:
Isn't it the same as `waypoint`?

Reply:
unfortunately no. Native waypoint doesn't transfer fields it can't recognise

Reply:
migrated to waypoints